data-pipeline/src/pipeline/pre_processing/ancillary/schools.py

```diff
@@ -93,3 +98,35 @@ def prepare_schools_data(base_data_path, links_data_path):
     return schools[(schools["Rank"] == 1) | (schools["Rank"].isna())].drop(
         columns=["LinkURN", "LinkName", "LinkType", "LinkEstablishedDate", "Rank"]
     )
+
+
+def _optional_ofsted_cols(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Ensure that "OfstedRating (name)" and "OfstedLastInsp" columns are present in the DataFrame,
+    even if they are missing from the original submission. Missing columns are created with defaults
+    to ensure compatibility with downstream processing.
+
+    These columns are required to write to the db and "OfstedRating (name)" is required for rag
+    calculations.
+
+    If the columns exist, they are either preserved or mapped as necessary.
+    If they do not exist, new columns are created: an empty string column for
+    "OfstedRating (name)" and a `None` column for "OfstedLastInsp".
+
+    :param df: The GIAS DataFrame to process.
```

Comment:
nitpick: we should be consistent with "gias" vs. "GIAS" where possible.

Personally, I'm in favour of the latter.